jekyll/_cci2/browser-testing.md

```diff
@@ -0,0 +1,239 @@
+*   [Automatically on failure, using Behat and Mink](https://gist.github.com/michalochman/3175175)
+
+### Using a Local Browser to Access HTTP server on CircleCI
+
+If you are running a test that runs an HTTP server on CircleCI, sometimes it is helpful to use a browser running on your local machine to debug a
+failing test. Setting this up is easy with an SSH-enabled
+run.
+
+First, run an SSH build using the Rerun Job with SSH button on the Builds page of the CircleCI app. The command to log into
+the container over SSH apears, as follows:
+
+```
+ssh -p 64625 ubuntu@54.221.135.43
+```
+
+To add port-forwarding to the command, use the `-L` flag.
+The following example forwards
+requests to `http://localhost:3000` to port `8080` on the CircleCI container.
+This would be useful, for example, if your build runs a debug Ruby on Rails app, which listens
+on port 8080.
+
+```
+ssh -p 64625 ubuntu@54.221.135.43 -L 3000:localhost:8080
+```
+
+Then, open your browser on your local machine and navigate to
+`http://localhost:8080` to send requests directly to the server
+running on port `3000` on the CircleCI container. You can also manually start the
+test server on the CircleCI container (if it is not already running), and you
+should be able to access the running test server from the browser on your development machine.
+
+This is a very easy way to debug things when setting up Selenium tests, for
+example.
+
+### Interacting With the Browser Over VNC
+
+VNC allows you to view and interact with the browser that is running your tests. This will only work if you're using a driver that runs a real browser. You will be able to interact with a browser that Selenium controls, but phantomjs is headless &mdash; there is nothing to interact with.
+
+Before you start, make sure you have a VNC viewer installed. If you're using macOS, we recommend
+[Chicken of the VNC](http://sourceforge.net/projects/chicken/).
+[RealVNC](http://www.realvnc.com/download/viewer/) is also available on most platforms.
+
+First, [start an SSH run]( {{ site.baseurl }}/2.0/ssh-access-jobs/)
+to a CircleCI container. When you connect to the machine, add the -L flag and forward the remote port 5901 to the local port 5902:
+
+```
+daniel@mymac$ ssh -p PORT ubuntu@IP_ADDRESS -L 5902:localhost:5901
+```
+
+You should be connected to the CircleCI container. Now start the VNC server:
+
+```
+ubuntu@box159:~$ vnc4server -geometry 1280x1024 -depth 24
+```
+
+Enter the password `password` when it prompts you. Your connection is secured with SSH, so there is no need for a strong password. You do need to enter a password to start the VNC server.
+
+Start your VNC viewer and connect to `localhost:5902`. Enter the password when it prompts you. You should see a display containing a terminal window. You can ignore any warnings about an insecure or unencrypted connection. Your connection is secured through the SSH tunnel.
+
+Next, make sure to run:
```

Comment:
What is so special about this step that we have to "make sure"? Especially when we use the word "ensure" below the command.

Change to: "Next, run:"

Reply:
thanks, I changed to 'be sure' int he step and changed ensure to 'enable windows to'